[BUG] Fix flaky test org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testReplicaThreadedUpdateToShardLimitsAndRejections

**Describe the bug**
New flaky test `org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testReplicaThreadedUpdateToShardLimitsAndRejections` ,  spotted in [1]

```
java.lang.AssertionError: expected null, but was:<org.opensearch.index.stats.IndexingPressurePerShardStats@5eae0090>
	at __randomizedtesting.SeedInfo.seed([9EFE9529CF42C133:9166A1D8EFC3EFA6]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotNull(Assert.java:756)
	at org.junit.Assert.assertNull(Assert.java:738)
	at org.junit.Assert.assertNull(Assert.java:748)
	at org.opensearch.index.ShardIndexingPressureConcurrentE
```

[1] https://build.ci.opensearch.org/job/gradle-check/1711/testReport/junit/org.opensearch.index/ShardIndexingPressureConcurrentExecutionTests/testReplicaThreadedUpdateToShardLimitsAndRejections/

**To Reproduce**
Steps to reproduce the behavior:
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testReplicaThreadedUpdateToShardLimitsAndRejections" -Dtests.seed=9EFE9529CF42C133 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=fr-LU -Dtests.timezone=America/Goose_Bay -Druntime.java=17
```

**Expected behavior**
Test should pass

**Plugins**
Standard

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - OS: CI
 - Version `main`

**Additional context**
Add any other context about the problem here.

--- a/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
+++ b/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
@@ -269,7 +269,6 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
 
         nodeStats = shardIndexingPressure.stats();
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
-        System.out.println("rejection count is: "+rejectionCount.get());
         if(rejectionCount.get() == NUM_THREADS){
             assertEquals(10, shardStoreStats.getCurrentPrimaryAndCoordinatingLimits());
         }

--- a/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
+++ b/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
@@ -269,6 +269,8 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
 
         nodeStats = shardIndexingPressure.stats();
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
+        // If rejection count equals NUM_THREADS that means rejections happened until the last request, then we'll get shardStoreStats which
+        // was updated on the last request. In other cases, the shardStoreStats simply moves to the cold store and null is returned.
         if (rejectionCount.get() == NUM_THREADS) {
             assertEquals(10, shardStoreStats.getCurrentPrimaryAndCoordinatingLimits());
         } else {
@@ -335,6 +337,8 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
         assertEquals(0, nodeStats.getCurrentReplicaBytes());
 
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
+        // If rejection count equals NUM_THREADS that means rejections happened until the last request, then we'll get shardStoreStats which
+        // was updated on the last request. In other cases, the shardStoreStats simply moves to the cold store and null is returned.
         if (rejectionCount.get() == NUM_THREADS) {
             assertEquals(15, shardStoreStats.getCurrentReplicaLimits());
         } else {

--- a/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
+++ b/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
@@ -269,7 +269,13 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
 
         nodeStats = shardIndexingPressure.stats();
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
-        assertNull(shardStoreStats);
+        System.out.println("rejection count is: "+rejectionCount.get());
+        if(rejectionCount.get() == NUM_THREADS){
+            assertEquals(10, shardStoreStats.getCurrentPrimaryAndCoordinatingLimits());
+        }
+        else{
+            assertNull(shardStoreStats);
+        }
         shardStats = shardIndexingPressure.coldStats();
         if (randomBoolean) {
             assertEquals(rejectionCount.get(), nodeStats.getCoordinatingRejections());
@@ -331,7 +337,12 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
         assertEquals(0, nodeStats.getCurrentReplicaBytes());
 
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
-        assertNull(shardStoreStats);
+        if(rejectionCount.get() == NUM_THREADS){
+            assertEquals(15, shardStoreStats.getCurrentReplicaLimits());
+        }
+        else{
+            assertNull(shardStoreStats);
+        }
 
         shardStats = shardIndexingPressure.coldStats();
         assertEquals(rejectionCount.get(), shardStats.getIndexingPressureShardStats(shardId1).getReplicaNodeLimitsBreachedRejections());

--- a/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
+++ b/server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java
@@ -269,10 +269,9 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
 
         nodeStats = shardIndexingPressure.stats();
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
-        if(rejectionCount.get() == NUM_THREADS){
+        if (rejectionCount.get() == NUM_THREADS) {
             assertEquals(10, shardStoreStats.getCurrentPrimaryAndCoordinatingLimits());
-        }
-        else{
+        } else {
             assertNull(shardStoreStats);
         }
         shardStats = shardIndexingPressure.coldStats();
@@ -336,10 +335,9 @@ public class ShardIndexingPressureConcurrentExecutionTests extends OpenSearchTes
         assertEquals(0, nodeStats.getCurrentReplicaBytes());
 
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
-        if(rejectionCount.get() == NUM_THREADS){
+        if (rejectionCount.get() == NUM_THREADS) {
             assertEquals(15, shardStoreStats.getCurrentReplicaLimits());
-        }
-        else{
+        } else {
             assertNull(shardStoreStats);
         }
 

